arch/arm/core/aarch32/cortex_m/fault.c

```diff
@@ -137,7 +137,9 @@ static void fault_show(const z_arch_esf_t *esf, int fault)
 
 #if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
 	PR_EXC("MMFSR: 0x%x, BFSR: 0x%x, UFSR: 0x%x",
-	       SCB_MMFSR, SCB_BFSR, SCB_UFSR);
+	       (SCB->CFSR & SCB_CFSR_MEMFAULTSR_Msk) >> SCB_CFSR_MEMFAULTSR_Pos,
+	       (SCB->CFSR & SCB_CFSR_BUSFAULTSR_Msk) >> SCB_CFSR_BUSFAULTSR_Pos,
+	       (SCB->CFSR & SCB_CFSR_USGFAULTSR_Msk) >> SCB_CFSR_USGFAULTSR_Pos);
```

Comment:
IMHO, it would be better to keep the existing `SCB_MMFSR`, `SCB_BFSR`, `SCB_UFSR` macros and re-define them in terms of the CMSIS field definitions, but no strong opinion.

Reply:
is there any established pattern in CMSIS I can look at?

Reply:
Well, you can take a look at the cmsis headers in the module

Reply:
I can only see static inline helpers in CMSIS, but nothing like what we had for `SCB_MMFSR`, etc. What I can do is:

```c
#define SCB_CFSR_MEMFAULTSTR \
	((SCB->CFSR & SCB_CFSR_MEMFAULTSR_Msk) >> SCB_CFSR_MEMFAULTSR_Pos)
```

@stephanosio ?

Reply:
@gmarull That looks reasonable.